sematic/resolvers/state_machine_resolver.py

```diff
@@ -97,28 +109,47 @@ def _set_future_state(self, future, state):
 
     # State machine lifecycle hooks
 
-    def _resolution_will_start(self) -> None:
+    def _resolution_will_start(self, root_future: AbstractFuture) -> None:
```

Comment:
Same here, `root_future` maybe not necessary.

Reply:
changed